BAZProjectC1/MovieAPI.swift

```diff
@@ -24,12 +24,126 @@ class MovieAPI {
         for result in results {
             if let id = result.object(forKey: "id") as? Int,
                let title = result.object(forKey: "title") as? String,
-               let poster_path = result.object(forKey: "poster_path") as? String {
-                movies.append(Movie(id: id, title: title, poster_path: poster_path))
+               let poster_path = result.object(forKey: "poster_path") as? String, let strDetail = result.object(forKey: "overview") as? String {
+                movies.append(Movie(id: id, title: title, descriptionMovie: strDetail, poster_path: poster_path))
             }
         }
-
         return movies
     }
+    
+    func getMoviesUpdate(completion: @escaping ([Movie]) -> ()) {
+        self.request(strUrlPath: "https://api.themoviedb.org/3/trending/movie/day?api_key=\(apiKey)", completion: { lstInfo in
+            completion(lstInfo)
+        })
+    }
+     
+    func getNowPlaying(completion: @escaping ([Movie]) -> ()) {
+        self.request(strUrlPath: "https://api.themoviedb.org/3/movie/now_playing?api_key=\(apiKey)", completion: { lstInfo in
+            completion(lstInfo)
+        })
+    }
+
+    func getMostPopular(completion: @escaping ([Movie]) -> ()) {
+        self.request(strUrlPath:"https://api.themoviedb.org/3/movie/popular?api_key=\(apiKey)&language=es&region=MX&page=1", completion: { lstInfo in
+            completion(lstInfo)
+        })
+    }
+    
+    func getTopRated(completion: @escaping ([Movie]) -> ()) {
+        self.request(strUrlPath:"https://api.themoviedb.org/3/movie/top_rated?api_key=\(apiKey)&language=es&region=MX&page=1", completion: { lstInfo in
+            completion(lstInfo)
+        })
+    }
+    
+    func getUpComing(completion: @escaping ([Movie]) -> ()) {
+        self.request(strUrlPath:"https://api.themoviedb.org/3/movie/upcoming?api_key=\(apiKey)&language=es&region=MX&page=1", completion: { lstInfo in
+            completion(lstInfo)
+        })
+    }
+    
+    func getQuery(strQuery: String, completion: @escaping ([Movie]) -> ()) {
+        guard let url = URL(string: "https://api.themoviedb.org/3/search/keyword?api_key=\(apiKey)&language=es&query=\(strQuery)") else {
+            return
+        }
+        
+        let session = URLSession.shared
+        session.dataTask(with: url) { (data, response, error) in
+            if let data = data {
```

Comment:
Simplifiquemos este código con un guard let
```suggestion
            guard let data = data else { completion([]); return }
```